cumulus/polkadot-parachain/polkadot-parachain-lib/src/nodes/manual_seal.rs

```diff
@@ -0,0 +1,234 @@
+		RuntimeApi: ConstructNodeRuntimeApi<Block, ParachainClient<Block, RuntimeApi>>,
+	> BuildImportQueue<Block, RuntimeApi> for BuildManualSealImportQueue<Block, RuntimeApi>
+{
+	fn build_import_queue(
+		client: Arc<ParachainClient<Block, RuntimeApi>>,
+		_block_import: ParachainBlockImport<Block, RuntimeApi>,
+		config: &Configuration,
+		_telemetry_handle: Option<TelemetryHandle>,
+		task_manager: &TaskManager,
+	) -> sc_service::error::Result<DefaultImportQueue<Block>> {
+		Ok(sc_consensus_manual_seal::import_queue(
+			Box::new(client.clone()),
+			&task_manager.spawn_essential_handle(),
+			config.prometheus_registry(),
+		))
+	}
+}
+
+pub struct ManualSealNode<NodeSpec>(PhantomData<NodeSpec>);
+
+impl<NodeSpec: NodeSpecT> BaseNodeSpec for ManualSealNode<NodeSpec> {
+	type Block = NodeSpec::Block;
+	type RuntimeApi = NodeSpec::RuntimeApi;
+	type BuildImportQueue = BuildManualSealImportQueue<Self::Block, Self::RuntimeApi>;
+}
+
+impl<NodeSpec: NodeSpecT> ManualSealNode<NodeSpec> {
+	pub fn new() -> Self {
+		Self(Default::default())
+	}
+
+	pub fn start_node<Net>(
+		&self,
+		config: Configuration,
+		para_id: ParaId,
+		block_time: u64,
+	) -> sc_service::error::Result<TaskManager>
+	where
+		Net: NetworkBackend<NodeSpec::Block, Hash>,
+	{
+		let PartialComponents {
+			client,
+			backend,
+			mut task_manager,
+			import_queue,
+			keystore_container,
+			select_chain: _,
+			transaction_pool,
+			other: (_, mut telemetry, _),
+		} = Self::new_partial(&config)?;
+		let select_chain = LongestChain::new(backend.clone());
+
+		let mut net_config = sc_network::config::FullNetworkConfiguration::<_, _, Net>::new(
+			&config.network,
+			config.prometheus_config.as_ref().map(|cfg| cfg.registry.clone()),
+		);
+		let metrics = Net::register_notification_metrics(
+			config.prometheus_config.as_ref().map(|cfg| &cfg.registry),
+		);
+
+		let syncing_strategy = build_polkadot_syncing_strategy(
+			config.protocol_id(),
+			config.chain_spec.fork_id(),
+			&mut net_config,
+			None,
+			client.clone(),
+			&task_manager.spawn_handle(),
+			config.prometheus_config.as_ref().map(|config| &config.registry),
+		)?;
+
+		let (network, system_rpc_tx, tx_handler_controller, start_network, sync_service) =
+			sc_service::build_network(sc_service::BuildNetworkParams {
+				config: &config,
+				client: client.clone(),
+				transaction_pool: transaction_pool.clone(),
+				spawn_handle: task_manager.spawn_handle(),
+				import_queue,
+				net_config,
+				block_announce_validator_builder: None,
+				syncing_strategy,
+				block_relay: None,
+				metrics,
+			})?;
+
+		let proposer = sc_basic_authorship::ProposerFactory::new(
+			task_manager.spawn_handle(),
+			client.clone(),
+			transaction_pool.clone(),
+			None,
+			None,
+		);
+
+		let (manual_seal_sink, manual_seal_stream) = futures::channel::mpsc::channel(1024);
+		let mut manual_seal_sink_clone = manual_seal_sink.clone();
+		task_manager.spawn_handle().spawn("block_authoring", None, async move {
```

Comment:
Nit: Could be essential, if for whatever reason sending to the sink fails we can crash.